frc_characterization/logger_analyzer/data_analyzer.py

```diff
@@ -233,106 +215,28 @@ def runAnalysis():
                 Tests.ARM: runAnalysisArm,
                 Tests.SIMPLE_MOTOR: runAnalysisSimple,
             }
+
+            self.prepared_data = self.prepare_data(
+                self.stored_data, window=self.window_size.get()
+            )
+
+            if not self.prepared_data["Valid"]:
+                return
```

Comment:
Also since the subsets need to be stored, I decided to just save it to a self.prepared_data dict.